external/twitter/src/main/scala/org/apache/spark/streaming/twitter/TwitterInputDStream.scala

```diff
@@ -86,12 +97,18 @@ class TwitterReceiver(
       })
 
       val query = new FilterQuery
-      if (filters.size > 0) {
-        query.track(filters.toArray)
-        newTwitterStream.filter(query)
-      } else {
-        newTwitterStream.sample()
+      query.count(count)
+      if (follow.size > 0) {
+        query.follow(follow.toArray)
+      }
+      if (track.size > 0) {
+        query.track(track.toArray)
+      }
+      if (locations.size > 0) {
+        query.locations(locations.flatMap(box =>
+          Seq(Array(box.west, box.south), Array(box.east, box.north))).toArray)
```

Comment:
Good catch. It won't compile without the `.toArray` at the end, but I can change the `Seq` to `Array`. I will update the code.